crates/bitwarden-crypto/src/store/context.rs

```diff
@@ -208,6 +247,156 @@ impl<Ids: KeyIds> KeyStoreContext<'_, Ids> {
+    pub fn wrap_private_key(
+        &self,
+        wrapping_key: Ids::Symmetric,
+        key_to_wrap: Ids::Asymmetric,
+    ) -> Result<EncString> {
+        let wrapping_key = self.get_symmetric_key(wrapping_key)?;
+        let private_key = self.get_asymmetric_key(key_to_wrap)?.to_owned();
+        private_key.to_der()?.encrypt_with_key(wrapping_key)
+    }
+
+    /// Decrypt and import a previously wrapped asymmetric private key into the context.
+    ///
+    /// The `wrapped_key` will be decrypted using `wrapping_key` and parsed as a PKCS#8
+    /// private key; the resulting key will be inserted as a local asymmetric key and the
+    /// new local identifier returned.
+    ///
+    /// # Errors
+    /// Returns an error if decryption or parsing fails.
+    pub fn unwrap_private_key(
+        &mut self,
+        wrapping_key: Ids::Symmetric,
+        wrapped_key: &EncString,
+    ) -> Result<Ids::Asymmetric> {
+        let wrapping_key = self.get_symmetric_key(wrapping_key)?;
+        let private_key_bytes: Vec<u8> = wrapped_key.decrypt_with_key(wrapping_key)?;
+        let private_key =
+            AsymmetricCryptoKey::from_der(&Pkcs8PrivateKeyBytes::from(private_key_bytes))?;
+        self.add_local_asymmetric_key(private_key)
+    }
+
+    /// Decrypt and import a previously wrapped signing key into the context.
+    ///
+    /// The wrapped COSE key will be decrypted with `wrapping_key` and parsed into a
+    /// `SigningKey` which is inserted as a local signing key. The new local identifier
+    /// is returned.
+    ///
+    /// # Errors
+    /// Returns an error if decryption or parsing fails.
+    pub fn unwrap_signing_key(
+        &mut self,
+        wrapping_key: Ids::Symmetric,
+        wrapped_key: &EncString,
+    ) -> Result<Ids::Signing> {
+        let wrapping_key = self.get_symmetric_key(wrapping_key)?;
+        let signing_key_bytes: Vec<u8> = wrapped_key.decrypt_with_key(wrapping_key)?;
+        let signing_key = SigningKey::from_cose(&CoseKeyBytes::from(signing_key_bytes))?;
+        self.add_local_signing_key(signing_key)
+    }
+
+    /// Return the verifying (public) key corresponding to a signing key identifier.
+    ///
+    /// This converts the stored `SigningKey` into a `VerifyingKey` suitable for
+    /// signature verification operations.
+    ///
+    /// # Errors
+    /// Returns an error if the signing key id does not exist.
+    pub fn get_verifying_key(&self, signing_key_id: Ids::Signing) -> Result<VerifyingKey> {
+        let signing_key = self.get_signing_key(signing_key_id)?;
+        Ok(signing_key.to_verifying_key())
+    }
+
+    /// Return the public key corresponding to an asymmetric (private) key identifier.
+    ///
+    /// This converts the stored private key into its public key representation.
+    ///
+    /// # Errors
+    /// Returns an error if the asymmetric key id does not exist.
+    pub fn get_public_key(
```

Comment:
Nice! I had a task to do something like this but never followed up on it: https://bitwarden.atlassian.net/browse/PM-18110

There's at least one use on `generate_user_fingerprint ` that could benefit from this, though not necessarily as part of this PR.